Mathlib/Analysis/Complex/Hadamard.lean

```diff
@@ -0,0 +1,488 @@
+
+/-- The function `F` is bounded above because `f` is. -/
+lemma F_BddAbove (hB : BddAbove ((norm ∘ f) '' (closedStrip 0 1))) :
+    BddAbove ((norm ∘ (F f)) '' (closedStrip 0 1)) := by
+ -- Rewriting goal
+  simp only [F, image_congr, comp_apply, map_mul, invInterpStrip]
+  rw [bddAbove_def] at *
+  cases' hB with B hB
+  -- Using bound
+  use ((max 1 ((sSupAbsIm f 0)^ (-(1 : ℝ)))) * max 1 ((sSupAbsIm f 1)^ (-(1 : ℝ)))) * B
+  simp only [mem_image, forall_exists_index, and_imp, forall_apply_eq_imp_iff₂]
+  intros z hset
+  specialize hB (norm (f z))
+  specialize hB _
+  · simp only [image_congr, mem_image, comp_apply]; use z
+  -- Proof that the bound is correct
+  · rw [norm_smul]
+    apply mul_le_mul _ hB (norm_nonneg _) _
+    · rw [norm_mul]
+      apply mul_le_mul _ _ (norm_nonneg _) (le_trans zero_le_one (le_max_left _ _))
+      -- Bounding individual terms
+      · by_cases hM0_one : 1 ≤ sSupAbsIm f 0
+        -- `1 ≤ (sSupAbsIm f 0)`
+        · apply le_trans _ (le_max_left _ _)
+          simp only [norm_eq_abs, abs_cpow_eq_rpow_re_of_pos h0, sub_re, one_re,
+          Real.rpow_le_one_of_one_le_of_nonpos hM0_one (sub_nonpos.mpr hset.2)]
+        -- `0 < (sSupAbsIm f 0) < 1`
+        · rw [not_le] at hM0_one; apply le_trans _ (le_max_right _ _)
+          simp only [norm_eq_abs, abs_cpow_eq_rpow_re_of_pos h0, sub_re, one_re]
+          apply Real.rpow_le_rpow_of_exponent_ge (h0) (le_of_lt hM0_one) _
+          simp only [neg_le_sub_iff_le_add, le_add_iff_nonneg_left, hset.1]
+      · by_cases hM1_one : 1 ≤ sSupAbsIm f 1
+        -- `1 ≤ sSupAbsIm f 1`
+        · apply le_trans _ (le_max_left _ _)
+          simp only [norm_eq_abs, abs_cpow_eq_rpow_re_of_pos h1, sub_re, one_re, neg_re,
+          Real.rpow_le_one_of_one_le_of_nonpos hM1_one (Right.neg_nonpos_iff.mpr hset.1)]
+        -- `0 < sSupAbsIm f 1 < 1`
+        · rw [not_le] at hM1_one; apply le_trans _ (le_max_right _ _)
+          simp only [norm_eq_abs, abs_cpow_eq_rpow_re_of_pos h1, sub_re, one_re, neg_re,
+          Real.rpow_le_rpow_of_exponent_ge (h1) (le_of_lt hM1_one)
+            (neg_le_neg_iff.mpr hset.2)]
+    simp only [gt_iff_lt, lt_max_iff, zero_lt_one, true_or, mul_nonneg_iff_of_pos_left, le_max_iff,
+      zero_le_one]
+
+-- Proof that the edges are bounded by one
+lemma F_edge_le_one (hB : BddAbove ((norm ∘ f) '' (closedStrip 0 1)))
+    (hz : z ∈ re ⁻¹' {0, 1}) : ‖F f z‖ ≤ 1 := by
+  simp only [F, invInterpStrip, norm_smul, norm_eq_abs, map_mul, abs_cpow_eq_rpow_re_of_pos,
+    h0, h1, sub_re, one_re, neg_re]
+  cases' hz with hz0 hz1
+  -- `z.re = 0`
+  · simp [hz0, mul_one, zero_sub, Real.rpow_zero, neg_zero,
+    Real.rpow_neg_one, inv_mul_le_iff h0]
+    rw [← hz0]
+    apply abs_le_sSupAbsIm f z _ hB
+    simp only [closedStrip, mem_preimage, zero_le_one, left_mem_Icc, hz0]
+  -- `z.re = 1`
+  · rw [mem_singleton_iff] at hz1
+    simp only [hz1, one_mul, Real.rpow_zero, sub_self, Real.rpow_neg_one,
+    inv_mul_le_iff h1, mul_one]
```

Comment:
```suggestion
      inv_mul_le_iff h1, mul_one]
```
When you have a multiline tactic like `simp` or `rw`, all lines but the first one are indented by two more spaces to help locate the beginning and the end of the tactic.